murano/api/v1/applications.py

```diff
@@ -0,0 +1,123 @@
+        return f(*args, **kwargs)
+
+    return f_normalize_path
+
+
+class Controller(object):
+    @request_statistics.stats_count(API_NAME, 'Index')
+    @utils.verify_template
+    @normalize_path
+    def index(self, request, template_id, path):
+        LOG.debug('Applications:Get <TemplateId: {0}, '
+                  'Path: {1}>'.format(template_id, path))
+
+        try:
+            result = core_services.CoreServices.get_template_data(template_id,
+                                                         path)
+        except (KeyError, ValueError, AttributeError):
+            raise exc.HTTPNotFound
+        return result
+
+    @request_statistics.stats_count(API_NAME, 'Show')
+    @utils.verify_template
+    @normalize_path
+    def show(self, request, template_id, path):
+        LOG.debug('Applications:Get <TemplateId: {0}, '
+                  'Path: {1}>'.format(template_id, path))
+
+        try:
+            result = core_services.CoreServices.get_template_data(template_id,
+                                                         path)
+            if result is None:
+                raise exc.HTTPNotFound
+        except (KeyError, ValueError, AttributeError):
+            raise exc.HTTPNotFound
+        return result
+
+    @request_statistics.stats_count(API_NAME, 'Create')
+    @utils.verify_template
+    @normalize_path
+    def post(self, request, template_id, path, body):
+        secure_data = token_sanitizer.TokenSanitizer().sanitize(body)
+        LOG.debug('Applications:Post <TempId: {0}, Path: {2}, '
+                  'Body: {1}>'.format(template_id, secure_data, path))
+
+        post_data = core_services.CoreServices.post_application_data
+        try:
+            result = post_data(template_id, body, path)
+        except (KeyError, ValueError):
+            raise exc.HTTPNotFound
+        return result
+
+    @request_statistics.stats_count(API_NAME, 'Update')
+    @utils.verify_template
+    @normalize_path
+    def put(self, request, template_id, path, body):
+        LOG.debug('Applications:Put <TempId: {0}, Path: {2}, '
+                  'Body: {1}>'.format(template_id, body, path))
+
+        put_data = core_services.CoreServices.put_data
+        session_id = request.context.session
+
+        try:
+            result = put_data(template_id, session_id, body, path)
+        except (KeyError, ValueError):
+            raise exc.HTTPNotFound
+        return result
+
+    @request_statistics.stats_count(API_NAME, 'Delete')
+    @utils.verify_template
+    @normalize_path
+    def delete(self, request, template_id, path):
+        LOG.debug('Applications:Put <TempId: {0}, '
+                  'Path: {1}>'.format(template_id, path))
+        delete_data = core_services.CoreServices.delete_template_data
+        try:
+            delete_data(template_id, path)
+        except (KeyError, ValueError):
+            raise exc.HTTPNotFound
+
+
+def create_resource():
```

Comment:
In general, I see several LOG.debug but no LOG.error or LOF.warning. Are you showing this information in upper class? If no, why are you not showing or logging it here?